File: maximum_product_difference_between_two_pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function maxProductDifference(nums) {
    let smallest = Infinity, secondSmallest = Infinity;
    let largest = -Infinity, secondLargest = -Infinity;
    
    for (const num of nums) {
        if (num <= smallest) {
            secondSmallest = smallest;
            smallest = num;
        } else if (num <= secondSmallest) {
            secondSmallest = num;
        }
        
        if (num >= largest) {
            secondLargest = largest;
            largest = num;
        } else if (num >= secondLargest) {
            secondLargest = num;
        }
    }
    
    return (largest * secondLargest) - (smallest * secondSmallest);
};

File: maximum_product_difference_between_two_pairs.cpp
class Solution {
    public:
   int maxProductDifference(vector<int>& nums) {
     // Your code goes here
     int max = INT_MIN;
     int sMax = INT_MIN;
     int min = INT_MAX;
     int sMin = INT_MAX;
    for(int num:nums){
        if(num>max){
            sMax = max;
            max=num;
        }else if(num>sMax){
            sMax=num;
        }
        if(num<min){
            sMin=min;
            min=num;
        }else if(num<sMin){
            sMin=num;
        }
    }
    return (max*sMax)-(min*sMin);
   }
};